src/main/java/de/tum/in/www1/artemis/domain/Submission.java

```diff
@@ -110,12 +112,48 @@ public Result getLatestResult() {
         return null;
     }
 
+    /**
+     * Used to get result by correction round
+     * @param correctionRound to get result by
+     * @return the result based on the given correction-round
+     */
+    @Nullable
+    @JsonIgnore
+    public Result getResultByCorrectionRound(Long correctionRound) {
+        if (results != null && results.size() > correctionRound) {
```

Comment:
Not sure but can results really be null since it's been initialized in the beginning

Reply:
If it can get `null` the methods below are missing checks.

Reply:
yep, you are right, results will never be null, only maybe empty. I removed that check.